test/lib/bamboo/mailer_test.exs

```diff
@@ -2,250 +2,182 @@ defmodule Bamboo.MailerTest do
     refute_received {:deliver, _, _}
   end
 
   test "deliver_later/1 with empty lists for recipients does not deliver email" do
-    new_email(to: [], cc: [], bcc: []) |> FooMailer.deliver_later()
+    new_email(to: [], cc: [], bcc: []) |> Mailer.deliver_later()
     refute_received {:deliver, _, _}
 
-    new_email(to: [], cc: nil, bcc: nil) |> FooMailer.deliver_later()
+    new_email(to: [], cc: nil, bcc: nil) |> Mailer.deliver_later()
     refute_received {:deliver, _, _}
 
-    new_email(to: nil, cc: [], bcc: nil) |> FooMailer.deliver_later()
+    new_email(to: nil, cc: [], bcc: nil) |> Mailer.deliver_later()
     refute_received {:deliver, _, _}
 
-    new_email(to: nil, cc: nil, bcc: []) |> FooMailer.deliver_later()
+    new_email(to: nil, cc: nil, bcc: []) |> Mailer.deliver_later()
     refute_received {:deliver, _, _}
   end
 
-  test "deliver_later/1 calls deliver on the adapter" do
-    email = new_email()
-
-    FooMailer.deliver_later(email)
-
-    assert_receive {:deliver, delivered_email, _config}
-    assert delivered_email == Bamboo.Mailer.normalize_addresses(email)
-  end
-
-  test "deliver_now/1 wraps the recipients in a list" do
-    address = {"Someone", "foo@bar.com"}
-    email = new_email(to: address, cc: address, bcc: address)
-
-    FooMailer.deliver_now(email)
-
-    assert_received {:deliver, delivered_email, _}
-    assert delivered_email.to == [address]
-    assert delivered_email.cc == [address]
-    assert delivered_email.bcc == [address]
-  end
-
-  test "deliver_now/1 converts binary addresses to %{name: name, email: email}" do
-    address = "foo@bar.com"
-    email = new_email(from: address, to: address, cc: address, bcc: address)
-
-    FooMailer.deliver_now(email)
-
-    converted_address = {nil, address}
-    assert_received {:deliver, delivered_email, _}
-    assert delivered_email.from == converted_address
-    assert delivered_email.to == [converted_address]
-    assert delivered_email.cc == [converted_address]
-    assert delivered_email.bcc == [converted_address]
-  end
-
-  test "converts structs with custom protocols" do
-    user = %Bamboo.Test.User{first_name: "Paul", email: "foo@bar.com"}
-    email = new_email(from: user, to: user, cc: user, bcc: user)
-
-    FooMailer.deliver_now(email)
-
-    converted_recipient = {user.first_name, user.email}
-    assert_received {:deliver, delivered_email, _}
-    assert delivered_email.from == {"#{user.first_name} (MyApp)", user.email}
-    assert delivered_email.to == [converted_recipient]
-    assert delivered_email.cc == [converted_recipient]
-    assert delivered_email.bcc == [converted_recipient]
-  end
-
   test "raises an error if an address does not have a protocol implemented" do
     email = new_email(from: 1)
 
     assert_raise Protocol.UndefinedError, fn ->
-      FooMailer.deliver_now(email)
+      Mailer.deliver_now(email)
     end
   end
 
-  test "raises if all recipients are nil" do
+  test "raises if all receipients are nil" do
```

Comment:
I think `recipients` is correct?

Reply:
Doh! Good catch! Not sure why I even changed that. Thanks.